phantazm-zombies-mapeditor/src/main/java/com/github/phantazmnetwork/zombies/mapeditor/client/ui/NewDoorGui.java

```diff
@@ -0,0 +1,61 @@
+package com.github.phantazmnetwork.zombies.mapeditor.client.ui;
+
+import com.github.phantazmnetwork.commons.LogicUtils;
+import com.github.phantazmnetwork.commons.Namespaces;
+import com.github.phantazmnetwork.commons.vector.Region3I;
+import com.github.phantazmnetwork.zombies.map.DoorInfo;
+import com.github.phantazmnetwork.zombies.map.MapInfo;
+import com.github.phantazmnetwork.zombies.mapeditor.client.EditorSession;
+import net.kyori.adventure.key.Key;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * GUI class used to create {@link DoorInfo}.
+ */
+public class NewDoorGui extends NamedObjectGui {
+    /**
+     * Constructs a new instance of this GUI. The user may create a new door, or add regions to an existing door.
+     * @param session the current {@link EditorSession}
+     */
+    @SuppressWarnings("PatternValidation")
+    public NewDoorGui(@NotNull EditorSession session) {
+        super(LogicUtils.nullCoalesce(session.lastDoor(), door -> door.id().value()));
+
+        Objects.requireNonNull(session, "session");
+
+        MapInfo currentMap = session.getMap();
+        Region3I selected = session.getSelection();
+        buttonAdd.setOnClick(() -> {
+            String value = textFieldName.getText();
+            if(value.isEmpty()) {
+                return;
+            }
+
+            Key doorKey = Key.key(Namespaces.PHANTAZM, value);
```

Comment:
should we be using phantazm keys for user-created objects?

Reply:
i mostly did it to make typing out the names easier, but it can be changed if necessary

Reply:
could we somehow make the namespace the mapname or something

Reply:
only reason we'd want different namespaces is if some other organization was making maps